x/contracts/runtime/contract.go

```diff
@@ -45,6 +45,9 @@ type CallInfo struct {
 	// the serialized parameters that will be passed to the called function
 	Params []byte
 
+	// true if the call passes all gas
+	PassAll bool
+
 	// the maximum amount of fuel allowed to be consumed by wasm for this call
 	Fuel uint64
```

Comment:
Would be interested to know if there could be a cleaner solution. Currently there is a bool and the uint64 that tells the amount of fuel passed, but since having `PassAll = true` and `Fuel > 0` is inconsistent state, this has to be validated through the execution. Is there a better way to simulate this rust enum:
```rust
enum Gas {
    PassAll,
    Units(u64)
}
```